include/fmt/format.h

```diff
@@ -628,6 +629,8 @@ FMT_CONSTEXPR inline auto utf8_decode(const char* s, uint32_t* c, int* e)
   *e |= uchar(s[3]) >> 6;
   *e ^= 0x2a;  // top two bits of each tail byte correct?
   *e >>= shifte[len];
+  *e |= ((uchar(s[0]) & prefix_masks[len]) !=
+         uchar((prefix_masks[len] << 1) & 0xFF));  // first byte correct?
```

Comment:
I don't think this is needed because `utf8_decode` already gives an error for "\xf0\x28...".

Reply:
Without this check failed at sequence ``\xf4\x8f\xbf\xc0``.
```
/Users/phprus/Devel/fmt/test/ranges-test.cc:393: Failure
Expected equality of these values:
  fmt::format("{}", vec{"\xf4\x8f\xbf\xc0"})
    Which is: "[\"\\xf4\\x0f\xBF\\xc0\"]"
  "[\"\\xf4\\x8f\\xbf\\xc0\"]"
```

``utf8_decode`` return zero in ``e``.

Reply:
Could you add a test case then?

Reply:
Nevermind, I missed the code snippet which shows it's already there.

Reply:
This test already present:
https://github.com/fmtlib/fmt/blob/541cd218381493728241fcc690e8c362434a37ea/test/ranges-test.cc#L383-L384

This patch fix two error.
First error fixed by add condition ``error ? 1 : to_unsigned(end - buf_ptr)``
Second - by add first byte check.

The first error masked the second on the sequence ``\xf4\x8f\xbf\xc0``

Reply:
AFAIC the current version of `utf8_decode` already returns an error for "\xf4\x8f\xbf\xc0" too: https://godbolt.org/z/E194qvshK

Reply:
Step 1:
``"\xf4\x8f\xbf\xc0"`` - is not valid utf-8, print ``\xf4``, remove ``\xf4`` from sequence.
Step 2:
``"\x8f\xbf\xc0\0"`` - is not valid utf-8, but utf8_decode return no error and invalid codepoint (15, not 143).

See: https://godbolt.org/z/EbeEex4Gf

Reply:
Makes sense, thanks.